### PR TITLE
feat: typed admin/return-reasons

### PR DESCRIPTION
**Pull request by kasperkristensen**

### CODE REVIEW

File: packages/medusa/src/api/routes/admin/return-reasons/index.ts [hunk 1]
```diff
@@ -45,7 +46,13 @@ export const defaultFields = [
   "deleted_at",
 ]
 
-export const defaultRelations = [
+export const defaultAdminReturnReasonsRelations = [
   "parent_return_reason",
   "return_reason_children",
 ]
+
+export type AdminReturnReasonsResponse = {
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/return-reasons/index.ts:54]:
```suggestion
export type AdminReturnReasonsRes = {
```

File: packages/medusa/src/api/routes/admin/return-reasons/list-reasons.ts [hunk 2]
```diff
@@ -20,13 +26,19 @@ import { defaultRelations, defaultFields } from "./"
...
+    select: defaultAdminReturnReasonsFields,
+    relations: defaultAdminReturnReasonsRelations,
   })
 
   res.status(200).json({ return_reasons: data })
 }
+
+export type AdminGetReturnReasonsResponse = {
+  return_reasons: ReturnReason[]
+}
```
Comment (srindom) [inline packages/medusa/src/api/routes/admin/return-reasons/list-reasons.ts:44]: Move to index - `AdminReturnReasonsListRes`

**Review by srindom — CHANGES_REQUESTED** [review]

File: packages/medusa/src/api/routes/admin/return-reasons/create-reason.ts [hunk 3]
```diff
@@ -1,11 +1,17 @@
...
 
 /**
  * @oas [post] /return-reasons
  * operationId: "PostReturnReasons"
  * summary: "Create a Return Reason"
  * description: "Creates a Return Reason"
+ * x-authenticated: true
  * requestBody:
  *   content:
  *     application/json:
```
Comment (pKorsholm) [inline packages/medusa/src/api/routes/admin/return-reasons/create-reason.ts:17]: remember to include the required fields(vaule, label) in the OAS documentation schema as 
```
schema: 
  required: 
    - value
    - label
```

**Review by olivermrbl — APPROVED** [review]
LGTM

**Pull request merged**